keps/sig-scheduling/3280-guarantee-pdb-when-preemption-happens/README.md

```diff
@@ -0,0 +1,973 @@
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+This design proposal suggests adding a field `AllowDisruptionByPriorityGreaterThanOrEqual` in the `PriorityClass` API 
+to explicitly indicate  that `PodDisruptionBudget` of the pods corresponding to this priorty class can only be 
+violated by pods with the priority value greater than or equal to the value of `AllowDisruptionByPriorityGreaterThanOrEqual` 
+during the scheduler preemption process, in order to meet the needs of users who do not want PDB violations to occur 
+during scheduler preemption due to some reasons such as high availability of 
+services(https://github.com/kubernetes/kubernetes/issues/91492#issuecomment-1029484252). 
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+`PodDisruptionBudget` (PDB) is used to limit the number of concurrent disruptions that your application experiences, 
+allowing for high availability. Users can set the field `.spec.maxUnavailable` or `.spec.minAvailable` to declare 
+the current minimum availability or maximum unavailability to be maintained after eviction. 
+
+However, there is currently an issue where the kube-scheduler does not strictly guarantee PDBs during the preemption 
+process. The scheduler supports PDBs when preempting pods, but the adherence to PDBs is best effort. The scheduler 
+attempts to select victims whose PDBs are not violated during preemption, but if no such victims are found, preemption 
+will still take place, resulting in the removal of lower-priority pods despite their PDBs being violated.
+
+PodDisruptionBudgets (PDBs) are frequently used for stability and the possibility of violating a PDB during the preemption 
+process is not acceptable for certain users. As such, it is beneficial to provide users with the option to choose if they 
+want the PDB to be guaranteed during preemption or not.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+- Provide an option to configure whether the scheduler needs to make `PodDisruptionBudget` 
```

Comment:
Maybe make explicit that this option is available to cluster administrators only.

And then clarify that it's a non goal to let application developers directly influence preemption behavior.

Reply:
Updated.